### PR TITLE
Referrals - Analytics

### PR DESCRIPTION
**Pull request by ashiagr**
## Description
This adds analytics events for Referrals.

### Share Guest Pass

1. Login with an account that has an Plus or Patron active
2. Go to Profile
3. You should see the referral tip/hint on top. Check event: `referral_tooltip_shown`
4. Tap on the gift icon on the top left. Check event: `referral_tooltip_tapped` and `referral_share_screen_show`
6. Tap on the Send Pass. Check event: `referral_share_screen_shown`
7. Save the URL

### Claim Guest Pass

1. Now login with an account that doesn't have a subscription
2. Open the referral URL
5. The PocketCasts app should open and show the claim guest pass screen.  Check event: `referral_claim_screen_shown`
6. Tap on Not Now. Check event: `referral_not_now_tapped`
7. Switch to Profile Tab if needed. You should see the claim banner on profile. Check event: `referral_pass_banner_shown`
8. Tap on the banner
10. Tap on Activate Pass. Check event: `referral_activate_tapped`
11. You should see a loading item on the button
12. Then you should see the IAP screen. Check event: `referral_purchase_shown`
13. Since you can not purchase the offer with the current setup, code-review that `referral_purchase_success` is tracked in `ReferralsClaimGuestPassViewModel` on purchase success.

## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [ ] I have considered whether it makes sense to add tests for my changes
- [ ] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [ ] Any jetpack compose components I added or changed are covered by compose previews
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.
 
#### I have tested any UI changes...

- [ ] with different themes
- [ ] with a landscape orientation
- [ ] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

File: modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/AnalyticsEvent.kt [hunk 1]
```diff
@@ -655,5 +655,14 @@ enum class AnalyticsEvent(val key: String) {
     TRANSCRIPT_SEARCH_PREVIOUS_RESULT("transcript_search_previous_result"),
 
     /* Referrals */
-    REFERRAL_LINK_SHARED("referral_link_shared"),
+    REFERRAL_TOOLTIP_SHOWN("referral_tooltip_shown"),
+    REFERRAL_TOOLTIP_TAPPED("referral_tooltip_tapped"),
+    REFERRAL_SHARE_SCREEN_SHOWN("referral_share_screen_shown"),
+    REFERRAL_PASS_SHARED("referral_pass_shared"),
+    REFERRAL_CLAIM_SCREEN_SHOWN("referral_claim_screen_shown"),
```
Comment (mebarbosa) [inline modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/AnalyticsEvent.kt:662]: We can close the claim screen by using the drag down gesture. Do you think we need to track this too?

**Review by mebarbosa — APPROVED** [review]
Nice work!

I left two comments, but since they are not blockers I am already approving the PR

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/AnalyticsEvent.kt (hunk 1 above)
Comment (ashiagr) [reply modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/AnalyticsEvent.kt:662]: Good catch!
According to the description, it should be tracked when `User taps the link on the claim screen "Not Now" or uses the back button`

I fixed it in de00a19.

File: modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/AnalyticsEvent.kt [hunk 2]
```diff
@@ -655,5 +655,14 @@ enum class AnalyticsEvent(val key: String) {
     TRANSCRIPT_SEARCH_PREVIOUS_RESULT("transcript_search_previous_result"),
 
     /* Referrals */
-    REFERRAL_LINK_SHARED("referral_link_shared"),
+    REFERRAL_TOOLTIP_SHOWN("referral_tooltip_shown"),
```
[the remark this answers is not in the document]
Comment (ashiagr) [reply modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/AnalyticsEvent.kt:658]: Asked here: p1727963725591689/1724650880.085479-slack-C07HDV91FBQ

[the remark this answers is not in the document]
Comment (ashiagr) [reply modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/AnalyticsEvent.kt:658]: Added in c44dfbe

**Pull request merged**